File: client/src/components/Analytics.js
// src/components/Analytics.js
import React from 'react';

const Analytics = ({ analytics }) => {
  return (
    <div>
      <h3>Analytics</h3>
      <p>Followers: {analytics.followers}</p>
      <p>Following: {analytics.following}</p>
      <p>Posts: {analytics.posts}</p>
      <p>Accounts Reached: {analytics.accountsReached}</p>
      <p>Accounts Engaged: {analytics.accountsEngaged}</p>
      <p>Ads Insights: {analytics.adsInsights}</p>
      <p>Active Promotions: {analytics.activePromotions}</p>
      <p>Total Stories: {analytics.totalStories}</p>
      <p>Total Follows: {analytics.totalFollows}</p>
      <p>Total Saves: {analytics.totalSaves}</p>
      <p>Total Comments: {analytics.totalComments}</p>
      <p>Total Shares: {analytics.totalShares}</p>
    </div>
  );
};

export default Analytics;

File: server/social-media-dashboard/src/main/java/com/example/social_media_dashboard/model/Analytics.java
package com.example.socialmediadashboard.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "analytics")
public class Analytics {
    @Id
    private String id;
    private int followers;
    private int following;
    private int posts;
    private int accountsReached;
    private int accountsEngaged;
    private int adsInsights;
    private int activePromotions;
    private int totalStories;
    private int totalFollows;
    private int totalSaves;
    private int totalComments;
    private int totalShares;
}
